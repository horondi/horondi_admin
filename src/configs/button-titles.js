const buttonTitles = {
  DELETE_TITLE: 'Видалити',
  EDIT_TITLE: 'Редагувати',
  CREATE_NEWS_TITLE: 'Додати новину',
  CREATE_PATTERN_TITLE: 'Додати гобелен',
  CREATE_MODEL_TITLE: 'Додати модель',
  MODEL_SAVE_TITLE: 'Зберегти',
  CANCEL_TITLE: 'Відмінити',
  LOGOUT_TITLE: 'Вихід',
  CREATE_BUSINESS_PAGE: 'Додати бізнес сторінку',
  CREATE_CONTACT_TITLE: 'Додати контакти',
  SWITCH_USER_STATUS_TITLE: 'Змінити статус користувача',
  USER_INACTIVE_TITLE: 'Деактивувати',
  USER_ACTIVE_TITLE: 'Активувати',
  ADD_CATEGORY: 'Додати категорію',
  GO_BACK_TITLE: 'Назад',
  ADD_SUBCATEGORY: 'Додати підкатегорію',
  ADD_CATEGORY_IMAGE: 'Додати зображення',
  ADD_CATEGORY_NAME: 'Додати назву',
  CANCEL: 'Відмінити',
  SAVE_CATEGORY: 'Зберегти категорію',
  SAVE_SUBCATEGORY: 'Зберегти підкатегорію',
  CREATE_SPECIAL_USER: 'Створити спецкористувача',
  ADD_PHOTO_LABEL: '+',
  CREATE_CATEGORY: 'Створити категорію',
  CREATE_SUBCATEGORY: 'Створити підкатегорію',
  PATTERN_REMOVE_MESSAGE: 'Ви впевнені, що хочете видалити цей гобелен?',
  REMOVE_CONTACT_MESSAGE: 'Ви впевнені, що хочете видалити цей контакт?',
  USER_UNACTIVE_TITLE: 'Деактивувати',
  SHOW_COMMENTS_TITLE: 'Переглянути коментарі',
  HIDE_COMMENTS_TITLE: 'Приховати коментарі',
  SAVE_ORDER: 'Зберегти замовлення',
  SAVE_TITLE: 'Зберегти',
  UNDERSTAND: 'Зрозуміло',
  ANSWER: 'Відповісти',
  TO_SPAM: 'У СПАМ',
  MOVE_ALL_TO_SPAM: 'Перемістити у СПАМ',
  CREATE_MATERIAL_TITLE: 'Створити матеріал',
  CREATE_COLOR_TITLE: 'Створити колір',
  SAVE_MATERIAL: 'Зберегти матеріал',
  CREATE_HEADER_TITLE: 'Створити посилання',
  HEADER_SAVE_TITLE: 'Зберегти посилання',
  GO_TO_MATERIAL_COLOR_PALLET: 'Редагувати палітру',
  CLOSE_DIALOG_TITLE: 'Закрити вікно',
  ORDER_DETAILS: 'Деталі',
  USER_STATUS_TITLE: 'Статус',
  CLEAR_FILTERS: 'Очистити фільтри',
  CREATE_SLIDE_TITLE: 'Створити слайд',
  OPEN_SLIDE_EDIT: 'Увімкнути редагування',
  SAVE_SLIDE_ORDER: 'Зберегти зміни',
  CANCEL_SLIDE_ORDER: 'Скасувати зміни',

  titleGenerator: (editMode, isMain) => {
    const editModeMap = new Map([
      [true, 'Зберегти'],
      [false, 'Створити']
    ]);
    const isMainMap = new Map([
      [true, 'категорію'],
      [false, 'підкатегорію']
    ]);
    return `${editModeMap.get(editMode)} ${isMainMap.get(isMain)}`;
  }
};
export default buttonTitles;
