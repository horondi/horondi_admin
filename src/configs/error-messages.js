export const errorMessages = {
  USER_NOT_FOUND: 'Користувач не знайдений!',
  USER_NOT_AUTHORIZED: 'Користувач не отримав прав доступу',
  INVALID_PERMISSIONS: 'Недостатньо прав користувача',
  WRONG_CREDENTIALS: 'Неправильно вказані вхідні дані',
  INPUT_NOT_VALID: 'Неправильні ввідні дані',
  USER_ALREADY_EXIST: 'Користувач з такими даними вже існує',
  INVALID_ADMIN_INVITATIONAL_TOKEN:
    'Неправильне посилання на створення користувача',
  NO_COLORS: 'Створіть колір'
};

export const loginErrorMessages = {
  INVALID_EMAIL_MESSAGE: 'Некоректна email адреса',
  ENTER_EMAIL_MESSAGE: 'Введіть email',
  PASSWORD_MIN_LENGTH_MESSAGE: 'Пароль повинен містити не менше 8 символів',
  PASSWORD_MAX_LENGTH_MESSAGE: 'Пароль повинен містити не більше 20 символів',
  PASSWORD_LANG_MESSAGE: 'Використовуйте латиницю різних регістрів та цифри',
  ENTER_FIRSTNAME_MESSAGE: "Введіть ім'я",
  ENTER_LASTNAME_MESSAGE: 'Введіть прізвище',
  ENTER_PASSWORD_MESSAGE: 'Введіть пароль',
  FIRSTNAME_MIN_LENGTH_MESSAGE: "Ім'я повинно містити не менше 2 символів",
  LASTNAME_MIN_LENGTH_MESSAGE: 'Прізвище повинно містити не менше 2 символів',
  FIRSTNAME_MAX_LENGTH_MESSAGE: "Ім'я повинно містити не більше 30 символів",
  LASTNAME_MAX_LENGTH_MESSAGE: 'Прізвище повинно містити не більше 30 символів',
  SELECT_ROLE_MESSAGE: 'Оберіть роль'
};
export const patternErrorMessages = {
  PATTERN_VALIDATION_ERROR: 'Мінімум 2 символи',
  PATTERN_ERROR_MESSAGE: 'Поле не може бути порожнім',
  PATTERN_ERROR_ENGLISH_AND_DIGITS_ONLY: 'Тільки англійські букви і цифри',
  PHOTO_NOT_PROVIDED: "Фото гобелена є обов'язковим",
  CONSTRUCTOR_PHOTO_NOT_PROVIDED: "Фото для конструктора є обов'язковим",
  PATTERN_UA_NAME_MESSAGE:
    'Поле може містити тільки українські літери та цифри',
  PATTERN_EN_NAME_MESSAGE: 'Поле може містити тільки англійські літери та цифри'
};
export const categoryErrorMessages = {
  CATEGORY_VALIDATION_ERROR: 'Мінімум 2 символи',
  CATEGORY_ERROR_MESSAGE: 'Поле не може бути порожнім',
  CATEGORY_UA_NAME_MESSAGE:
    'Поле може містити тільки українські літери та цифри',
  CATEGORY_EN_NAME_MESSAGE:
    'Поле може містити тільки англійські літери та цифри',
  CATEGORY_CODE_MESSAGE: 'Поле не може містити спеціальні символи'
};
export const headerErrorMessages = {
  HEADER_VALIDATION_ERROR: 'Мінімум 2 символи',
  HEADER_ERROR_MESSAGE: 'Поле не може бути порожнім',
  HEADER_ERROR_ENGLISH_AND_DIGITS_ONLY: 'Тільки англійські букви і цифри'
};
export const newsErrorMessages = {
  NAME_MIN_LENGTH_MESSAGE: `Ім'я автора повинне містити не менше 2 символів`,
  TITLE_MIN_LENGTH_MESSAGE: 'Заголовок повинен містити не менше 10 символів',
  TEXT_MIN_LENGTH_MESSAGE: 'Текстове поле повино містити не менше 10 символів'
};
export const modelErrorMessages = {
  MODEL_VALIDATION_ERROR: 'Мінімум 2 символи',
  MODEL_ERROR_MESSAGE: 'Поле не може бути порожнім',
  MODEL_ERROR_ENGLISH_AND_DIGITS_ONLY: 'Тільки англійські букви і цифри',
  PHOTO_NOT_PROVIDED: 'Додайте фото для моделі'
};
export const contactErrorMessages = {
  INVALID_EMAIL_MESSAGE: 'Некоректна email адреса',
  ENTER_EMAIL_MESSAGE: 'Введіть email',
  PHONE_NUMBER_LENGTH_MESSAGE:
    'Довжина номеру телефону повинна містити 12 символів',
  PHONE_NUMBER_TYPE_MESSAGE: 'Номер повинен містити лише числа',
  ENTER_PHONE_NUMBER_MESSAGE: 'Введіть номер',
  INPUT_LENGTH_MESSAGE: 'Довжина повинна містити не менше 10 символів',
  ENTER_UA_SCHEDULE_MESSAGE: 'Введіть розклад українською',
  ENTER_EN_SCHEDULE_MESSAGE: 'Введіть розклад англійською',
  ENTER_UA_ADDRESS_MESSAGE: 'Введіть адресу українською',
  ENTER_EN_ADDRESS_MESSAGE: 'Введіть адресу англійською',
  IMAGE_FORMAT_MESSAGE:
    'Введіть коректний формат, наприклад: https://example.com/',
  ENTER_LINK_MESSAGE: 'Введіть посилання',
  SELECT_IMAGES_MESSAGE: 'Завантажте зображення для карт'
};

export const emailQuestionsErrorMessages = {
  ANSWER_INPUT_MESSAGE: 'Введіть текст для відповіді'
};

export const materialErrorMessages = {
  MAX_LENGTH_MESSAGE: `Не більше 300 символів`,
  MIN_LENGTH_MESSAGE: `Не менше 2 символів`,
  VALIDATION_ERROR: 'Поле обовязкове',
  PRICE_VALIDATION_ERROR: 'Тільки цифри'
};
export const sizeErrorMessages = {
  MAX_LENGTH_MESSAGE: `Не більше 20 символів`,
  MIN_LENGTH_MESSAGE: `Не менше 1см`,
  MAX_LENGTH_MESSAGE_SIZE: `Не більше 35см`,
  MIN_LENGTH_MESSAGE_SIZE: `Не менше 1 символу`,
  MAX_WEIGHT_MESSAGE_SIZE: `Не більше 5 кг`,
  MIN_WEIGHT_MESSAGE_SIZE: `Не менше 0.1 кг`,
  VALIDATION_ERROR: `Поле обовязкове`,
  PRICE_VALIDATION_ERROR: `Тільки цифри`,
  NOT_UA_INPUT_MESSAGE: `Введіть ім'я розміру українською`,
  NOT_EN_INPUT_MESSAGE: `Введіть ім'я розміру англійською`,
  NO_NUMBER_TYPE_MESSAGE: `Повинно містити число`,
  NO_STRING_TYPE_MESSAGE: `Повинно містити букви`
};

export const colorErrorMessages = {
  MAX_LENGTH_MESSAGE: `Не більше 100 символів`,
  MIN_LENGTH_MESSAGE: `Не менше 2 символів`,
  COLOR_VALIDATION_ERROR: 'Неправильний формат кольору',
  VALIDATION_ERROR: 'Поле обовязкове'
};
export const statsErrorMessages = {
  NO_STATS: 'Статистика для вибраного значення відсутня'
};
export const homePageSlideErrorMessages = {
  SLIDE_VALIDATION_ERROR: 'Мінімум 2 символи',
  SLIDE_ERROR_MESSAGE: 'Поле не може бути порожнім',
  SLIDE_ERROR_ENGLISH_AND_DIGITS_ONLY: 'Тільки англійські букви і цифри'
};

export const constructorErrorMessages = {
  CONSTRUCTOR_VALIDATION_ERROR: 'Мінімум 2 символи',
  CONSTRUCTOR_ERROR_MESSAGE: 'Поле не може бути порожнім',
  CONSTRUCTOR_ERROR_ENGLISH_AND_DIGITS_ONLY: 'Тільки англійські букви і цифри',
  PRICE_VALIDATION_ERROR: 'Тільки цифри',
  PHOTO_NOT_PROVIDED: 'Додайте фото для конструктора'
};
export const paginationInputErrorMessages = {
  MUST_BE_NUMBER: 'Тільки цифри',
  MUST_BE_POSITIVE: 'Тільки більше нуля',
  PAGE_NOT_FOUND: 'Сторінку не знайдено'
};

export const commentErrorMessages = {
  COMMENT_VALIDATION_ERROR: 'Мінімум 2 символи',
  COMMENT_ERROR_MESSAGE: 'Поле не може бути порожнім',
  MAX_LENGTH_MESSAGE: 'Не більше 300 символів'
};
