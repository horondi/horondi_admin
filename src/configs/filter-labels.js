const filterLabels = {
  categories: {
    sortLabels: [
      {
        value: 'sortByNameAsc',
        label: "за ім'ям (у алфавітному порядку)",
        key: 'name',
        type: 'asc'
      },
      {
        value: 'sortByNameDesc',
        label: "за ім'ям (проти алфавітного порядку)",
        key: 'name',
        type: 'desc'
      }
    ]
  },
  orders:{
    users: [
      {
        value: 'sortByNameAsc',
        label: "за ім'ям (у алфавітному порядку)",
        key: 'name',
        type: 'asc'
      },
      {
        value: 'sortByNameDesc',
        label: "за ім'ям (проти алфавітного порядку)",
        key: 'name',
        type: 'desc'
      },
      {
        value: 'sortByEmailAsc',
        label: 'за поштою (у алфавітному порядку)',
        key: 'email',
        type: 'asc'
      },
      {
        value: 'sortByEmailDesc',
        label: 'за поштою (проти алфавітного порядку)',
        key: 'email',
        type: 'desc'
      }
    ],
    sortLabels:[
      {
        value: 'sortByTotalPriceToPayAsc',
        label: "за вартістю товару (по зростанню)",
        key: 'totalPriceToPay.0.value',
        type: 'asc'
      },
      {
        value: 'sortByTotalPriceToPayDesc',
        label: "за вартістю товару (по спаданню)",
        key: 'totalPriceToPay.0.value',
        type: 'desc'
      },
      {
        value: 'sortByCreatedAtAsc',
        label: "від старих до нових",
        key: 'dateOfCreation',
        type: 'asc'
      },
      {
        value: 'sortByCreatedDesc',
        label: "від нових до старих",
        key: 'dateOfCreation',
        type: 'desc'
      },
    ]
  }
};

export default filterLabels;
