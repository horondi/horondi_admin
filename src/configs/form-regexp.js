const formRegExp = {
  patternMaterial: '^[A-Za-z][A-Za-z0-9]*$',
  email:
    '^([\\w-]+(?:\\.[\\w-]+)*)@((?:[\\w-]+\\.)*\\w[\\w-]{0,66})\\.([a-z]{2,6}(?:\\.[a-z]{2})?)$',
  pass: '^(?!.* )(?=.*[0-9])(?=.*[A-Z]).{8,30}$',
  unwrapHtml: /(<([^>]+)>)/gi,
  enAddressRegex: '^[A-Za-z0-9_|,| |./]+$',
  onlyPositiveDigits: /^[1-9]\d*$/,
  editorField: /^<p><br><\/p>|<p><\/p>$/,
  mobileNumber: /^\+380\(\d{2}\)-\d{3}-\d{2}-\d{1,2}$/g,
  userName: /[а-яА-Я]{2,}/g,
  userRoles: /(Користувач|Адмін|Суперадмін)/g,
  userStatuses: /(Активний|Неактивний)/g,
  uaNameCreation: /^[а-яїієґ0-9]+$/i,
  enNameCreation: /^[a-z0-9]+$/i,
  categoryCode: /^[a-z0-9|-]/i
};
export default formRegExp;
