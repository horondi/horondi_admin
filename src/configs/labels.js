const labels = {
  sort: 'Сортувати за',
  search: 'Шукати',
  user: {
    unknownAdmin: 'Невідомий адмін'
  },
  model: {
    name: 'Назва',
    description: 'Опис',
    image: 'Фото моделі',
    category: 'Категорія',
    available: 'Доступний',
    show: 'Показати',
    avatarText: 'Фото',
    priority: 'Пріорітет',
    showEnable: 'Так',
    showDisable: 'Ні'
  },
  categories: {
    switchCategory: "Категорія, на яку замінити пов'язані товари",
  },
  header: {
    name: 'Назва',
    link: 'Посилання',
    priority: 'Пріорітет'
  },
  pattern: {
    image: 'Фото гобелена',
    material: 'Код матеріалу',
    available: 'Доступний',
    handmade: 'Зроблений вручну',
    avatarText: 'Фото'
  },
  material: {
    image: 'Фото матеріалу',
    purpose: 'Застосування',
    available: 'Доступний',
    name: 'Назва матеріалу',
    description: 'Опис матеріалу',
    additionalPrice: 'Додаткова ціна'
  },
  colors: {
    image: 'Фото кольору',
    name: 'Назва кольору',
    simpleName: 'Проста назва кольору',
    code: 'Код кольору',
    available: 'Доступний'
  },
  product: {
    sortBySelectOptions: [
      {
        label: 'популярністю',
        value: 'popularity'
      },
      {
        label: 'від дорогих до дешевих',
        value: 'sortDesc'
      },
      {
        label: 'від дешевих до дорогих',
        value: 'sortAsc'
      },
      {
        label: 'рейтингом',
        value: 'rate'
      }
    ],
    stepsLabels: [
      'Введіть інформацію про продукт',
      'Оберіть категорію, підкатегорію, модель, колір, гобелен та ціну продукту',
      'Вкажіть ціну продукту',
      'Оберіть опційні параметри',
      'Завантажте фото для продукту',
      'Підтвердження створення продукту'
    ],
    infoLabels: [
      { label: 'Назва', name: 'name', required: true },
      {
        label: 'Основний матеріал',
        name: 'mainMaterial',
        required: true
      },
      {
        label: 'Внутрішній матеріал',
        name: 'innerMaterial',
        required: false
      },
      { label: 'Замок', name: 'closure', required: false },
      {
        label: 'Довжина лямок(см)',
        name: 'strapLengthInCm',
        required: false
      },
      { label: 'Опис', name: 'description', required: false }
    ],
    selectsLabels: [
      {
        label: 'Категорія ',
        name: 'category',
        type: 'select',
        required: true
      },
      {
        label: 'Модель ',
        name: 'model',
        type: 'select',
        required: true
      },
      {
        label: 'Колір ',
        name: 'colors',
        type: 'select',
        required: true
      },
      {
        label: 'Гобелен ',
        name: 'pattern',
        type: 'select',
        required: true
      }
    ],
    optionsLabels: [
      { label: 'Розміри', name: 'sizes' },
      { label: 'Нижні матеріали', name: 'bottomMaterials' }
    ],
    sizeCardsLabels: [
      { label: 'Розмір', name: 'name' },
      { label: `Об'єм(л)`, name: 'volumeInLiters' },
      { label: 'Ширина(см)', name: 'widthInCm' },
      { label: 'Висота(см)', name: 'heightInCm' },
      { label: 'Глибина(см)', name: 'depthInCm' }
    ],
    materialsLabels: [{ label: `Назва`, name: 'name' }],
    optionsValues: {
      sizes: [],
      bottomMaterials: [],
      additions: false
    },
    priceLabel: {
      label: 'Ціна(USD) ',
      name: 'basePrice'
    }
  },
  orderProduct: {
    category: 'Категорія',
    subcategory: 'Підкатегорія',
    model: 'Модель',
    name: 'Назва',
    pattern: 'Гобелен',
    closure: 'Матеріал замку',
    bottomMaterial: 'Матеріал дна',
    bottomColor: 'Колір дна',
    actualPrice: 'Ціна'
  },
  orderRecipient: {
    firstName: "Ім'я",
    lastName: 'Прізвище',
    patronymicName: 'По-батькові',
    email: 'e-mail',
    phoneNumber: 'Номер телефону',
    commentary: 'Коментар'
  },
  deliveryDetails: {
    country: 'Країна',
    region: 'Область',
    city: 'Місто',
    zipcode: 'Поштовий індекс',
    street: 'Вулиця',
    buildingNumber: 'Номер будинку',
    appartment: 'Номер квартири'
  },
  sizeValues: {
    heightInCm: 'Висота (см.)',
    widthInCm: 'Ширина (см.)',
    depthInCm: 'Глибина (см.)',
    volumeInLiters: "Об'єм (л.)",
    weightInKg: 'Вага (кг.)'
  },
  emailQuestionsLabels: {
    en: {
      ALL: 'ALL',
      PENDING: 'PENDING',
      SPAM: 'SPAM',
      ANSWERED: 'ANSWERED'
    },
    ua: {
      ALL: 'Всі',
      PENDING: 'Очікує відповіді',
      SPAM: 'Спам',
      ANSWERED: 'Відповідь надано'
    },
    placeholder: 'Відповідь ...'
  },
  doughnut: {
    select: [
      { label: 'Популярні категорії за весь час', value: 'categories' },
      { label: 'Останні замовлення', value: 'orders' }
    ],
    dateMenuOptions: [
      { label: 'За 7 Днів', value: 7 },
      { label: 'За 14 Днів', value: 14 },
      { label: 'За місяць', value: 30 },
      { label: 'За 3 місяці', value: 90 },
      { label: 'За рік', value: 365 }
    ]
  },
  bar: {
    select: [
      { label: 'Популярні продукти за весь час', value: 'products' },
      { label: 'Виконані замовлення', value: 'orders' },
      { label: 'Останні зареєстровані користувачі', value: 'users' }
    ],
    descriptions: {
      users: 'Цей день',
      orders: 'Цей день',
      products: 'Кількість покупок'
    },
    message: {
      users: 'Кількість користувачів за цей час: ',
      orders: 'Кількість замовлень за цей час: ',
      products: 'Кількість продуктів за цей час: '
    }
  },
  orderTabs: {
    general: 'Загальне',
    receiver: 'Отримувач',
    products: 'Продукти',
    delivery: 'Доставка'
  },
  deliveryLabels: {
    deliveryMethodLabel: 'Спосіб доставки',
    byCourierLabel: "Доставка кур'єром:",
    invoiceNumberLabel: 'Номер накладної',
    warehouseNumberLabel: 'Номер відділення: ',
    sentAtLabel: 'Відправлено о:',
    deliveryCostLabel: 'Вартість доставки'
  },
  generalLabels: {
    deliveryStatusLabel: 'Статус замовлення:',
    paymentMethodLabel: 'Метод оплати:',
    isPaidLabel: 'Оплачено:',
    cancellationReasonLabel: 'Причина скасування',
    creationDateLabel: 'Дата створення:',
    updateDateLabel: 'Дата оновлення:',
    adminCommentLabel: 'Залишити коментар'
  },
  productsLabels: {
    notListed: 'Не вказано',
    additionsLabel: 'Додатки',
    colorsLabel: 'Кольори',
    sizeLabel: 'Розмір'
  },
  orders: {
    select: [
      { label: 'Створені', value: 'CREATED' },
      { label: 'Виконані', value: 'CONFIRMED' },
      { label: 'Оброблені', value: 'PRODUCED' },
      { label: 'Скасовані', value: 'CANCELLED' },
      { label: 'Повернені', value: 'REFUNDED' },
      { label: 'Надіслані', value: 'SENT' },
      { label: 'Доставлені', value: 'DELIVERED' },
      { label: 'Всі', value: 'All' }
    ]
  },
  homePageSlide: {
    image: 'Фото слайду',
    title: 'Заголовок слайду',
    show: 'Доступний',
    description: 'Опис слайду',
    link: 'Посилання для переходу'
  }
};
export default labels;
