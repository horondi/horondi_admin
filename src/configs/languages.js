const languages = ['ua', 'en'];

export default languages;