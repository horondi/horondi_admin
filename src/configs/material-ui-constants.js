const materialUiConstants = {
  outlinedAgeNativeSimple: 'outlined-age-native-simple',
  typographyVariantH1: 'h1',
  contained: 'contained',
  outlined: 'outlined',
  secondary: 'secondary',
  primary: 'primary',
  save: 'save',
  show: 'show',
  off: 'off',
  types: {
    number: 'number',
    string: 'string',
    submit: 'submit'
  },
  codeError: 'code-error'
};

export default materialUiConstants;
