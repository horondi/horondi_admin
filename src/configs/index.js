import { colors } from '@material-ui/core';
import routes from './routes';
import menuCategories from './menu-categories';
import tableHeadRowTitles from './table-head-row-titles';
import detailTitles from './detail-titles';
import buttonTitles from './button-titles';
import statuses from './statuses';
import {
  loginErrorMessages,
  colorErrorMessages,
  contactErrorMessages,
  patternErrorMessages,
  materialErrorMessages,
  newsErrorMessages,
  errorMessages,
  modelErrorMessages,
  headerErrorMessages,
  homePageSlideErrorMessages,
  categoryErrorMessages
} from './error-messages';
import messages from './messages';
import formRegExp from './form-regexp';
import labels from './labels';
import {
  newsPaginationPayload,
  contactsPaginationPayload,
  materialPaginationPayload
} from './pagination';
import titles from './titles';
import { sort } from './sort';

export const config = {
  app: {
    title: 'Horondi Admin Portal',
    serverUrl: 'http://localhost:5000/',
    drawerWidth: 230,
    snackBarDuration: 4000,
    rowsPerPageOptions: [10, 25, 50, 100]
  },
  languages: ['uk', 'en'],
  tokenName: 'HORONDI_AUTH_TOKEN',
  templates: {
    categoryTemplate: {
      available: false,
      code: '',
      images: {
        large: '',
        medium: '',
        small: '',
        thumbnail: ''
      },
      isMain: false,
      name: [],
      subcategories: []
    }
  },
  userRoles: [
    { role: 'user', label: 'Юзер' },
    { role: 'admin', label: 'Адмін' },
    { role: 'superadmin', label: 'Суперадмін' }
  ],
  allowedforRegistrationRoles: ['admin'],
  tabNames: {
    users: ['Користувачі', 'Адміністратори']
  },
  tableSizes: {
    SMALL_SIZE: 'small',
    DEFAULT_SIZE: 'medium'
  },
  iconSizes: {
    SMALL_SIZE: 'small',
    DEFAULT_SIZE: 'default'
  },
  theme: {
    DARK_THEME: 'dark',
    LIGHT_THEME: 'light'
  },
  forbiddenRolesFromDeleting: ['superadmin'],
  dialogFormTitles: {
    REGISTER_ADMIN: 'Створити спецкористувача'
  },
  paginationPayload: {
    skip: 0,
    limit: 5,
    countPerPage: 6
  },
  IMG_URL: 'https://horondi.blob.core.windows.net/horondi/images/',
  newsPerPage: 6,
  product: {
    sortBySelectOptions: [
      {
        label: 'популярністю',
        value: 'popularity'
      },
      {
        label: 'від дорогих до дешевих',
        value: 'sortDesc'
      },
      {
        label: 'від дешевих до дорогих',
        value: 'sortAsc'
      },
      {
        label: 'рейтингом',
        value: 'rate'
      }
    ],
    responsive: {
      superLargeDesktop: {
        breakpoint: { max: 4000, min: 3000 },
        items: 1
      },
      desktop: {
        breakpoint: { max: 3000, min: 1024 },
        items: 1
      },
      tablet: {
        breakpoint: { max: 1146, min: 464 },
        items: 1
      },
      mobile: {
        breakpoint: { max: 810, min: 0 },
        items: 1
      }
    }
  },
  popularity: 'popularity',
  rate: 'rate',
  sortAsc: 'sortAsc',
  sortDesc: 'sortDesc',
  submitKey: 'Enter',
  imagePrefix: 'https://horondi.blob.core.windows.net/horondi/images/',
  initialLanguageValues: [
    {
      lang: 'uk',
      value: ''
    },
    {
      lang: 'en',
      value: ''
    }
  ],
  doughnut: {
    colors: [
      colors.green[500],
      colors.red[700],
      colors.amber[600],
      colors.indigo[500]
    ],
    initialValues: {
      names: [],
      counts: [],
      relations: [],
      total: null
    }
  },
  bar: {
    initialValues: {
      labels: [],
      counts: []
    }
  },
  UAH: 'грн',
  buttonTitles,
  detailTitles,
  messages,
  menuCategories,
  routes,
  statuses,
  tableHeadRowTitles,
  formRegExp,
  labels,
  titles,
  loginErrorMessages,
  colorErrorMessages,
  contactErrorMessages,
  patternErrorMessages,
  materialErrorMessages,
  newsErrorMessages,
  categoryErrorMessages,
  errorMessages,
  newsPaginationPayload,
  contactsPaginationPayload,
  materialPaginationPayload,
  modelErrorMessages,
  headerErrorMessages,
  homePageSlideErrorMessages,
  sort
};
export const inputTypes = {
  button: 'button',
  submit: 'submit'
};
