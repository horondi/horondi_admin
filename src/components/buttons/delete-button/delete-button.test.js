import React from 'react';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import DeleteButton from './delete-button';
import { config } from '../../../configs';

Enzyme.configure({ adapter: new Adapter() });

describe('delete button tests', () => {
  const size = 'small';
  const id = '5f62f5386d3d7c14710c0111';
  const mockCallBack = jest.fn(() => id);
  const text = 'Delete';
  let component;
  beforeEach(() => {
    component = mount(
      <DeleteButton size={size} onClickHandler={mockCallBack} />
    );
  });
  afterEach(() => {
    component.unmount();
  });
  it('should exist', () => {
    expect(component).toMatchSnapshot();
    expect(component.exists());
  });
  it('should have type', () => {
    expect(component.find('button').type()).toEqual('button');
    expect(component).toMatchSnapshot();
  });
  it('should click', () => {
    expect(mockCallBack.mock.calls.length).toBe(0);
    component.simulate('click');
    expect(component.find('button'));
  });
  it('should exist and have value', () => {
    expect(component.exists());
    expect(component.prop('size')).toEqual(size);
    expect(component.prop('onClickHandler')).toEqual(mockCallBack);
  });
  it('should have props', () => {
    expect(component.props().size).toBeDefined();
    expect(component.props().size).toEqual(size);
    expect(component.props().onClickHandler).toBeDefined();
    expect(component.props().onClickHandler).toBeTruthy();
  });
  it('should have default props', () => {
    expect(DeleteButton.propTypes.size).toBeDefined();
    expect(DeleteButton.propTypes.onClickHandler).toBeDefined();
  });
  it('should be one', () => {
    expect(component.find('button').length).not.toBe(2);
  });
  it('should renders children when passed in', () => {
    const component = shallow(
      <DeleteButton>
        <div className='unique' />
      </DeleteButton>
    );
    expect(component.contains(<div className='unique' />)).toEqual(true);
  });
  it('should have title text (children)', () => {
    const component = shallow(<DeleteButton>{text}</DeleteButton>);
    expect(component.text()).toBeDefined();
    expect(component.text()).toEqual('Delete');
    expect(component.text()).toBeTruthy();
  });
  it('should have className', () => {
    const component = shallow(<DeleteButton />);
    expect(component.hasClass('makeStyles-button-8')).toBeDefined();
    expect(component.hasClass('makeStyles-button-8')).toBe(true);
    expect(component).toMatchSnapshot();
  });
});
