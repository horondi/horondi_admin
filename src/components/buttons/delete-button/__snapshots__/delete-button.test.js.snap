// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`delete button tests should exist 1`] = `ReactWrapper {}`;
